app/services/change_course.rb

```diff
@@ -0,0 +1,41 @@
+class ChangeCourse
+  include ImpersonationAuditHelper
+
+  attr_reader :actor, :application_choice, :course_option
+
+  def initialize(actor:,
+                 application_choice:,
+                 course_option:)
+    @actor = actor
+    @application_choice = application_choice
+    @course_option = course_option
+  end
+
+  def save!
+    auth.assert_can_make_decisions!(application_choice: application_choice, course_option: course_option)
+
+    if course.valid?
+      audit(actor) do
+        ActiveRecord::Base.transaction do
+          application_choice.update_course_option_and_associated_fields!(
+            course_option,
+            other_fields: { course_option: course_option },
```

Comment:
hmm might be worth adding a timestamp for changed course at 🤔 let's discuss tomorrow

Reply:
are we holding off on this, for this PR?

Reply:
yes